docs/advanced.md

```diff
@@ -0,0 +1,71 @@
+---
+id: advanced
+title: Advanced state management with DSL
+sidebar_label: DSL
+---
+
+Setting the response body as described in the [previous section](basic.md) is often sufficient, but sometimes you need to more control to test, for example, the behaviour for status code 500 or start with 10 users in a service. This can be achieved with the unmock DSL. All DSL elements are prefixed with the dollar sign (`$`).
+
+## Top-level DSL
+
+Top-level DSL is not specific to any property in the response body and can therefore only be found at the top level of the object passed as a state. At any other level, it will be treated as a proper key. All of the commands below can apply to the top-level DSL or to a specific mocked service.
```

Comment:
This should probably be expanded. The idea for `top level DSL` does not apply to e.g. `$size` - as it can be applied to any `array` element in a response (or the entire response, if it is an array). Originally I had separated this two, Mike later joined them. I'm not sure that makes it easier to understand.
At any case, that's for another PR.

Reply:
👍  I did not really understand the top-level DSL section, it needs work.

Reply:
I agree. I was trying to convey the idea that "top level DSL" is more "meta" than other items. So, it describes stuff about the response/state in general, and not specific elements within the response. As such, it only makes sense as a "top level" definition (or, alternatively, as another argument, which I think we agreed was not too appealing?).

So e.g. `{ $code: XXX, ... }` and `{ $times: N }` are considered DSL instructions, where as `{ items: { id: { $times: N } } }` is just a state (`$times` has no special meaning when it's nested like that).